cunumeric/deferred.py

```diff
@@ -1657,6 +1663,54 @@ def _diag_helper(
 
         task.execute()
 
+    @auto_convert([1, 2])
+    def put(self, indices: Any, values: Any) -> None:
+        if indices.base.kind == Future or indices.base.transformed:
+            indices = indices._convert_future_to_regionfield(
+                indices.base.kind == Future
+            )
```

Comment:
I don't think you need to do that, since `indices` itself is not used in a copy

Reply:
I believe I need to do that when indices is a scalar that was converted to ndarray, e.g it has shape `()`. See above